src/passes/J2CLOpts.cpp

```diff
@@ -105,7 +105,9 @@ class ConstantHoister : public WalkerPass<PostWalker<ConstantHoister>> {
       // TODO: maybe we should not introduce "nop" in the first place and try
       // removing instead.
       PassRunner runner(getModule());
+      runner.add("precompute");
       runner.add("vacuum");
+      runner.add("remove-unused-brs");
```

Comment:
Is there a benefit to putting `remove-unused-brs` last? In general `vacuum` is used that way (but it is not a strict rule).

Reply:
hmm.  It definitely has an effect. I extracted this out of a bigger change and what I noticed is in the bigger change I see 

```
(func $getXXX (...) 
  (return (global.get $global))
)
```

without "remove-unused-br" and

```
(func $getXXX (...) 
  (global.get $global)
)
```
if it is in.

Not really necessary though. In my bigger change I was collecting methods that had only `(global.get ...)`  but I can collect `(return (global.get ...))`  and achieve the same effect.

Reply:
Oh, sorry, I do agree `--remove-unused-brs` is useful. It removes `returns` like that. But can it run before the vacuum?

Reply:
I misunderstood. Yes it can run before. I'll update the pr.